pkg/purser_plugin/costing.go

```diff
@@ -222,3 +223,37 @@ func GetAllNodesCost() {
 	printNodeDetails(nodes)
 	//fmt.Println(nodes)
 }
+
+/*
+GetGroupCost returns Cost (total, cpu, memory and storage) for a Group
+*/
+func GetGroupCost(group *crd.Group) Cost {
+	cpuCostPerCPUPerHour, memCostPerGBPerHour, storageCostPerGBPerHour := GetUserCosts()
+	cost := Cost{}
+	cost.cpuCost = 0.0
+	cost.memoryCost = 0.0
+	cost.storageCost = 0.0
+
+	podsDetails := group.Spec.PodsDetails
+	for podName, podDetails := range podsDetails {
+		startTime := podDetails.StartTime
+		endTime := podDetails.EndTime
+
+		podTime := currentMonthActiveTimeInHours(startTime, endTime)
+
+		podMetrics := group.Spec.PodsMetrics[podName]
+		podCPU := resourceQuantityToFloat64(podMetrics.CpuRequest)
+		podMem := resourceQuantityToFloat64(podMetrics.MemoryRequest)
+
+		// TODO: find podStorage
+		podStorage := 0.0
+
+		cost.cpuCost += cpuCostPerCPUPerHour * podCPU * podTime
+		cost.memoryCost += memCostPerGBPerHour * podMem * podTime
+		cost.storageCost += storageCostPerGBPerHour * podStorage * podTime
+	}
+
+	cost.totalCost = cost.cpuCost + cost.memoryCost + cost.storageCost
+
+	return cost
```

Comment:
To improve the code readability, we can directly return the struct as: 

```
return &Cost{
     cpuCost:           totalCPUCost,
     memoryCost:   totalMemoryCost,
     storageCost:    totalStrorageCost,
     totalCost:         totalCumulativeCost,
}
```

This also eliminates the need to declare and initialise the struct separately in the beginning. 

Also, to update the struct field values iteratively use can consider creating separate functions like `addCPUCost()` as can be seen in this [blog](https://www.ardanlabs.com/blog/2017/06/for-range-semantics.html).

Reply:
Cost struct is used by other functions too. So I will change all of them in another PR.

Reply:
[Just a suggestion] 
I recommend fixing it here for this PR and fixing the other occurrences as part of another PR.